ludwig/backend/__init__.py

```diff
@@ -97,3 +98,14 @@ def initialize_backend(backend):
         backend = create_backend(backend)
     backend.initialize()
     return backend
+
+
+@contextlib.contextmanager
+def provision_preprocessing_workers(backend):
+    if backend.BACKEND_TYPE == RAY:
+        backend.provision_preprocessing_workers()
```

Comment:
Maybe better to have a default no-op implementation of provision/release workers, then override only in the RayBackend.

Reply:
I tried that out at first but it adds so much code for a very specialized case. I wanted to minimize code liter. A good compromise might be to just add it in the base class that raises a NotImplemented Error. What do you think?

Reply:
Sounds reasonable :)